python/pyiceberg/table/partitioning.py

```diff
@@ -178,4 +176,10 @@ def assign_fresh_partition_spec_ids(spec: PartitionSpec, old_schema: Schema, fre
                 transform=field.transform,
             )
         )
-    return PartitionSpec(INITIAL_SPEC_ID, fields=tuple(partition_fields))
+    if len(spec.fields) > 0:
+        return PartitionSpec(*partition_fields, spec_id=INITIAL_PARTITION_SPEC_ID)
+    return UNPARTITIONED_PARTITION_SPEC
+
+
+UNPARTITIONED_PARTITION_SPEC_ID = 0
```

Comment:
I don't think we should have this constant, since we can't rely on it.

Reply:
And it is gone